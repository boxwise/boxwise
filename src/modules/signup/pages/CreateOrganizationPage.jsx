import React from "react";
import { withRouter } from "react-router-dom";
import Grid from "@material-ui/core/Grid";

import { Page } from "modules/layout/components";
import AppBarHeader from "components/AppBarHeader";

import CreateOrganizationFlow from "../components/CreateOrganizationFlow";

export const CreateOrganizationPage = ({ history }) => (
  <React.Fragment>
    <AppBarHeader backUrl="/signin" title="Set up Boxwise" />
    <Page>
      <Grid container spacing={24} justify="center" alignItems="center">
        <Grid item xs={12} md={4}>
          <br />
          <br />
          <CreateOrganizationFlow
            onDone={() => {
              history.push("/");
            }}
          />
        </Grid>
      </Grid>
    </Page>
  </React.Fragment>
);

export default withRouter(CreateOrganizationPage);
