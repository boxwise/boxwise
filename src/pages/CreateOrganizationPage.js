import React from "react";
import { withRouter, Link } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import ArrowBack from "@material-ui/icons/ArrowBack";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";

import Page from "components/Page";
import CreateOrganizationFlow from "components/CreateOrganizationFlow";

const styles = theme => ({
  titleText: {
    [theme.breakpoints.up("sm")]: {
      margin: "0 auto"
    }
  }
});

export const CreateOrganizationPage = ({ classes, history }) => (
  <div>
    <AppBar position="fixed">
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="go back"
          component={Link}
          to="/signin"
        >
          <ArrowBack />
        </IconButton>
        <Typography
          className={classes.titleText}
          variant="h6"
          color="inherit"
          noWrap
        >
          {"Set up Boxwise"}
        </Typography>
      </Toolbar>
    </AppBar>
    <Toolbar />
    <Page>
      <Grid container spacing={24} justify="center" alignItems="center">
        <Grid item xs={12} md={4}>
          <br />
          <br />
          <CreateOrganizationFlow
            onDone={() => {
              history.push("/");
            }}
          />
        </Grid>
      </Grid>
    </Page>
  </div>
);

export default withStyles(styles)(withRouter(CreateOrganizationPage));
