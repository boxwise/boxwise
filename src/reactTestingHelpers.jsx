import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, fireEvent } from "react-testing-library";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import thunk from "redux-thunk";
import { createStore, applyMiddleware } from "redux";

export { fireEvent, wait } from "react-testing-library";

// following suggestions in https://testing-library.com/docs/react-testing-library/setup#custom-render

export const typeText = (element, value) => {
  fireEvent.change(element, { target: { value } });
};

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  }
});

export function customRender(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState = {},
    reducer = state => state
  } = {}
) {
  const store = createStore(reducer, initialState, applyMiddleware(thunk));
  return render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <Router history={history}>{ui}</Router>
      </Provider>
    </MuiThemeProvider>
  );
}

export { customRender as render };
