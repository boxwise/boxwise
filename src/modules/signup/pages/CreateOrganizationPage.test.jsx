import React from "react";

import { render } from "reactTestingHelpers";

import { CreateOrganizationPage } from "./CreateOrganizationPage";

it("renders correctly", () => {
  const { container } = render(<CreateOrganizationPage />);
  expect(container).toMatchSnapshot();
});
