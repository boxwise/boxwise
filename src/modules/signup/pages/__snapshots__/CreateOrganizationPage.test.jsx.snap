// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <WithStyles(AppBarHeader)
    backUrl="/signin"
    title="Set up Boxwise"
  />
  <WithStyles(Page)>
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      justify="center"
      spacing={24}
    >
      <WithStyles(Grid)
        item={true}
        md={4}
        xs={12}
      >
        <br />
        <br />
        <CreateOrganizationFlow
          onDone={[Function]}
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Page)>
</Fragment>
`;
